Probe Helper Fails With Fetching OAuth2 Token

**Describe the bug**
probe_helper_test.go fails with oauth2: cannot fetch token: 400 Bad Request

 **Expected behavior**
Unit tests should not need to fetch auth tokens.

Yes, this is the test output on master:

```
--- FAIL: TestProbeHelper (0.97s)
    logger.go:130: 2020-09-22T17:41:08.372Z     INFO    probe_helper/probe_helper.go:627        Starting Probe Helper server...
    logger.go:130: 2020-09-22T17:41:08.372Z     INFO    probe_helper/probe_helper.go:631        Starting event receiver...
    logger.go:130: 2020-09-22T17:41:08.373Z     INFO    probe_helper/probe_helper.go:103        Starting the probe helper health checker...
    logger.go:130: 2020-09-22T17:41:08.374Z     INFO    probe_helper/probe_helper.go:190        Received probe request{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: broker-e2e-delivery-probe
          source: probe-helper-test
          id: broker-e2e-delivery-probe-1234567890
          time: 2020-09-22T17:41:08.372639983Z
        }
    logger.go:130: 2020-09-22T17:41:08.375Z     INFO    probe_helper/probe_helper.go:285        Received event{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: broker-e2e-delivery-probe
          source: probe-helper-test
          id: broker-e2e-delivery-probe-1234567890
          time: 2020-09-22T17:41:08.372639983Z
        }
    logger.go:130: 2020-09-22T17:41:08.376Z     INFO    probe_helper/probe_helper.go:190        Received probe request{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: broker-e2e-delivery-probe
          source: probe-helper-test
          subject: invalid-subject
          id: broker-e2e-delivery-probe-1234567890
          time: 2020-09-22T17:41:08.372641578Z
        }
    logger.go:130: 2020-09-22T17:41:08.376Z     INFO    probe_helper/probe_helper.go:156        Probe forwarding failed, unrecognized probe event type and subject: type=broker-e2e-delivery-probe, subject=invalid-subject
    logger.go:130: 2020-09-22T17:41:08.376Z     INFO    probe_helper/probe_helper.go:190        Received probe request{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: cloudpubsubsource-probe
          source: probe-helper-test
          id: cloudpubsubsource-probe-1234567890
          time: 2020-09-22T17:41:08.372642904Z
        }
    logger.go:130: 2020-09-22T17:41:08.394Z     INFO    probe_helper/probe_helper.go:285        Received event{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: google.cloud.pubsub.topic.v1.messagePublished
          source: //pubsub.googleapis.com/projects/test-project-id/topics/cloudpubsubsource-topic
          id: m0
          time: 2020-09-22T17:41:08.387767328Z
          dataschema: https://raw.githubusercontent.com/googleapis/google-cloudevents/master/proto/google/events/cloud/pubsub/v1/data.proto
          datacontenttype: application/json
        Data,
          {
            "subscription": "cre-src-test-subscription-id",
            "message": {
              "messageId": "m0",
              "data": null,
              "attributes": {
                "ce-id": "cloudpubsubsource-probe-1234567890",
                "ce-source": "probe-helper-test",
                "ce-specversion": "1.0",
                "ce-time": "2020-09-22T17:41:08.372642904Z",
                "ce-type": "cloudpubsubsource-probe"
              },
              "publishTime": "2020-09-22T17:41:08.387767328Z"
            }
          }
        }
    logger.go:130: 2020-09-22T17:41:08.394Z     INFO    probe_helper/probe_helper.go:190        Received probe request{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: cloudpubsubsource-probe
          source: probe-helper-test
          subject: invalid-subject
          id: cloudpubsubsource-probe-1234567890
          time: 2020-09-22T17:41:08.372644224Z
        }
    logger.go:130: 2020-09-22T17:41:08.394Z     INFO    probe_helper/probe_helper.go:156        Probe forwarding failed, unrecognized probe event type and subject: type=cloudpubsubsource-probe, subject=invalid-subject
    logger.go:130: 2020-09-22T17:41:08.395Z     INFO    probe_helper/probe_helper.go:190        Received probe request{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: cloudstoragesource-probe
          source: probe-helper-test
          subject: create
          id: cloudstoragesource-probe-1234567890
          time: 2020-09-22T17:41:08.37264563Z
        }
    logger.go:130: 2020-09-22T17:41:08.472Z     INFO    probe_helper/probe_helper.go:285        Received event{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: google.cloud.scheduler.job.v1.executed
          source: //cloudscheduler.googleapis.com/test-cloud-scheduler-source
          id: 1234567890
        }
    logger.go:130: 2020-09-22T17:41:08.473Z     INFO    probe_helper/probe_helper.go:156        This event is not received by the probe receiver client:  
    logger.go:130: 2020-09-22T17:41:08.521Z     INFO    probe_helper/probe_helper.go:156        Probe forwarding failed, error closing storage writer for object cloudstoragesource-probe-1234567890: Post "http://127.0.0.1:38395/upload/storage/v1/b/cloudstoragesource-bucket/o?alt=json&name=cloudstoragesource-probe-1234567890&prettyPrint=false&projection=full&uploadType=multipart": oauth2: cannot fetch token: 400 Bad Request
        Response: {
          "error": "invalid_grant",
          "error_description": "reauth related error (invalid_rapt)",
          "error_subtype": "invalid_rapt"
        }
    --- FAIL: TestProbeHelper/CloudStorageSource_probe (0.13s)
        probe_helper_test.go:509: wanted result , got 500: 
    logger.go:130: 2020-09-22T17:41:08.522Z     INFO    probe_helper/probe_helper.go:190        Received probe request{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: cloudstoragesource-probe
          source: probe-helper-test
          subject: invalid-subject
          id: cloudstoragesource-probe-1234567890
          time: 2020-09-22T17:41:08.372649548Z
        }
    logger.go:130: 2020-09-22T17:41:08.522Z     INFO    probe_helper/probe_helper.go:156        Probe forwarding failed, unrecognized probe event type and subject: type=cloudstoragesource-probe, subject=invalid-subject
    logger.go:130: 2020-09-22T17:41:08.522Z     INFO    probe_helper/probe_helper.go:190        Received probe request{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: cloudauditlogssource-probe
          source: probe-helper-test
          id: cloudauditlogssource-probe-1234567890
          time: 2020-09-22T17:41:08.37265056Z
        }
    logger.go:130: 2020-09-22T17:41:08.572Z     INFO    probe_helper/probe_helper.go:285        Received event{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: google.cloud.scheduler.job.v1.executed
          source: //cloudscheduler.googleapis.com/test-cloud-scheduler-source
          id: 1234567890
        }
    logger.go:130: 2020-09-22T17:41:08.573Z     INFO    probe_helper/probe_helper.go:156        This event is not received by the probe receiver client:  
    logger.go:130: 2020-09-22T17:41:08.573Z     INFO    probe_helper/probe_helper.go:285        Received event{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: google.cloud.audit.log.v1.written
          source: //cloudaudit.googleapis.com/projects/test-project-id/logs/activity
          subject: pubsub.googleapis.com/projects/test-project-id/topics/cloudauditlogssource-probe-1234567890
          id: 1234567890
        Extensions,
          methodname: google.pubsub.v1.Publisher.CreateTopic
        }
    logger.go:130: 2020-09-22T17:41:08.574Z     INFO    probe_helper/probe_helper.go:190        Received probe request{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: cloudauditlogssource-probe
          source: probe-helper-test
          subject: invalid-subject
          id: cloudauditlogssource-probe-1234567890
          time: 2020-09-22T17:41:08.372651627Z
        }
    logger.go:130: 2020-09-22T17:41:08.574Z     INFO    probe_helper/probe_helper.go:156        Probe forwarding failed, unrecognized probe event type and subject: type=cloudauditlogssource-probe, subject=invalid-subject
    logger.go:130: 2020-09-22T17:41:08.574Z     INFO    probe_helper/probe_helper.go:190        Received probe request{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: cloudschedulersource-probe
          source: probe-helper-test
          id: cloudschedulersource-probe-1234567890
          time: 2020-09-22T17:41:08.372652493Z
        }
    logger.go:130: 2020-09-22T17:41:08.575Z     INFO    probe_helper/probe_helper.go:190        Received probe request{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: cloudschedulersource-probe
          source: probe-helper-test
          subject: invalid-subject
          id: cloudschedulersource-probe-1234567890
          time: 2020-09-22T17:41:08.37265325Z
        }
    logger.go:130: 2020-09-22T17:41:08.575Z     INFO    probe_helper/probe_helper.go:156        Probe forwarding failed, unrecognized probe event type and subject: type=cloudschedulersource-probe, subject=invalid-subject
    logger.go:130: 2020-09-22T17:41:08.575Z     INFO    probe_helper/probe_helper.go:190        Received probe request{event 25 0  Validation: valid
        Context Attributes,
          specversion: 1.0
          type: unrecognized-probe-type
          source: probe-helper-test
          id: unrecognized-probe-type-1234567890
          time: 2020-09-22T17:41:08.372654208Z
        }
    logger.go:130: 2020-09-22T17:41:08.575Z     INFO    probe_helper/probe_helper.go:156        Probe forwarding failed, unrecognized probe event type and subject: type=unrecognized-probe-type, subject=
FAIL
FAIL    github.com/google/knative-gcp/test/test_images/probe_helper     2.627s
FAIL
```

--- a/test/test_images/probe_helper/probe_helper_test.go
+++ b/test/test_images/probe_helper/probe_helper_test.go
@@ -305,7 +305,7 @@ func testStorageClient(ctx context.Context, t *testing.T) (*storage.Client, chan
 		gotRequest <- r
 		w.Write([]byte("{}"))
 	}))
-	c, err := storage.NewClient(ctx, option.WithEndpoint(srv.URL))
+	c, err := storage.NewClient(ctx, option.WithoutAuthentication(), option.WithEndpoint(srv.URL))
 	if err != nil {
 		t.Fatalf("Failed to create test storage client: %v", err)
 	}

